### ISSUE TITLE
[project reference add using] Reference desktop class library from Phone 8 project crashes codefix 

### ISSUE THREAD

**Issue by jmarolf** [dotnet/roslyn#8785]
1. Create Phone 8 project app
2. Create Desktop class library `Class1`
3. Add code that will reference `var a = new Class1();`

**Expected**: Add Using is not offered because it is invalid to reference Desktop from Phone 8

**Actual**: Codefix crashes with following stack *and* blocks the UI attempting to add the using to the file 

![image](https://cloud.githubusercontent.com/assets/9797472/13096617/8232628a-d4cf-11e5-9bc5-9e8ac16ef0ca.png)

```
User canceled out of save dialog (Exception from HRESULT: 0x8004000C (OLE_E_PROMPTSAVECANCELLED))
 at VSLangProj.References.AddProject(Project pProject)
 at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioWorkspaceImpl.ApplyProjectReferenceAdded(ProjectId projectId, ProjectReference projectReference)
 at Microsoft.CodeAnalysis.Workspace.ApplyProjectChanges(ProjectChanges projectChanges)
 at Microsoft.CodeAnalysis.Workspace.TryApplyChanges(Solution newSolution)
 at Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioWorkspaceImpl.TryApplyChanges(Solution newSolution)
 at Microsoft.CodeAnalysis.CodeActions.ApplyChangesOperation.Apply(Workspace workspace, CancellationToken cancellationToken)
 at Microsoft.CodeAnalysis.Editor.Implementation.CodeActions.CodeActionEditHandlerService.ProcessOperations(Workspace workspace, Document fromDocument, String title, Solution oldSolution, Solution updatedSolution, List`1 operationsList, CancellationToken cancellationToken)
 at Microsoft.CodeAnalysis.Editor.Implementation.CodeActions.CodeActionEditHandlerService.Apply(Workspace workspace, Document fromDocument, IEnumerable`1 operations, String title, CancellationToken cancellationToken)
 at Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.<InvokeWorkerAsync>d__18.MoveNext() --- End of stack trace from previous location where exception was thrown ---
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
 at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
 at Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.<>c__DisplayClass17_0.<<InvokeCoreAsync>b__0>d.MoveNext() --- End of stack trace from previous location where exception was thrown ---
 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
 at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
 at Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.<PerformActionAsync>d__2.MoveNext()
```

**NOTE**: The add using operation never completes unless the user clicks *cancel*.

**Comment by jmarolf** [comment]
@CyrusNajmabadi

**Comment by CyrusNajmabadi** [comment]
Looks to be a dupe of your other bug.  But i'm glad we have both to try out.

**Comment by jmarolf** [comment]
@CyrusNajmabadi as per **NOTE** The reason I included this is that the ui locks up as well, so their may be something else going on.

**Comment by davkean** [comment]
This was fixed in #8832.
